src/components/ListItem.jsx

```diff
@@ -1,5 +1,60 @@
 import './ListItem.css';
+import { updateItem } from '../api/firebase.js';
+import { useState, useEffect, useMemo } from 'react';
+import { ONE_DAY_IN_MILLISECONDS } from '../utils/dates.js';
 
-export function ListItem({ name }) {
-	return <li className="ListItem">{name}</li>;
+export function ListItem({ listPath, item }) {
+	/* Returns a boolean that is passed into isChecked useState
+	On render, box is checked if purchased less than a day ago */
+
+	const purchasedOneDayAgo = useMemo(() => {
+		if (item.dateLastPurchased !== null) {
+			const timeDiff = Date.now() - item.dateLastPurchased.seconds * 1000;
+			if (timeDiff <= ONE_DAY_IN_MILLISECONDS) {
+				return true;
+			} else {
+				return false;
+			}
+		} else {
+			return false;
+		}
+	}, [item.dateLastPurchased]);
+
+	const [isChecked, setIsChecked] = useState(purchasedOneDayAgo);
+	function changeHandler(e) {
+		setIsChecked(!isChecked);
+		async function purchaseItem() {
+			try {
+				await updateItem(listPath, item.id, !isChecked);
+			} catch (error) {
+				alert(error.message);
+			}
+		}
+		purchaseItem();
+	}
+
+	//sets a timer to uncheck an item 24 hours after it's purchased
+	useEffect(() => {
+		const timer = setTimeout(() => {
+			if (purchasedOneDayAgo) {
+				setIsChecked(false);
+			}
+		}, ONE_DAY_IN_MILLISECONDS);
+		return () => clearTimeout(timer);
+	}, [isChecked, purchasedOneDayAgo]);
+
```

Comment:
Question: does this timer rely upon local storage / the app staying up on our clients computer in order for it to tick down the full 24 hours worth of milliseconds? Wondering if it is interrupted when a user closes the app / shuts down their computer, etc.

Reply:
yes, it would reset if you refresh the page, so if you want to change the timeout time you'd have to first calculate how much time elapsed since the item was checked (so `currentTime - dateLastPurchased`), and then subtract from one day (`ONE_DAY_IN_MILLISECONDS - elapsedTime`)

Reply:
It makes sense to account for users re-opening a page before the 24 hours is over. Right now, the checkbox displays as unchecked only if the page is refreshed after the 24 hours has passed, or 24 hours after the page has been open. Definitely something to add to the list of things to refactor!

Reply:
I couldn't wait.... refactored the code to change the timeout time.<img src="https://media1.giphy.com/media/zyin7TYoGmLAs/giphy.gif"/>

Reply:
Awesome! Looks good to me, really quick handling of that particular concern.

Reply:
Agreed, that refactoring looks great!